src/main/java/com/puppycrawl/tools/checkstyle/CheckProperty.java

```diff
@@ -0,0 +1,14 @@
+package com.puppycrawl.tools.checkstyle;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation indicates that the field is a check property.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.FIELD)
+public @interface CheckProperty {
```

Comment:
I am curious on why you choose an annotation for each type of module instead of a singular one like `ModuleProperty`? Also what about TreeWalker property? Technically users can add listeners, and I believe they can have properties too, so should they be added? What about `Checker`, it isn't a Check nor FileFilter?

They are all properties, and we would know if they are for a check or filter based on it's class' heirarchy.

Reply:
I was thinking of manually annotating properties in `Checker`, `AbstractCheck`, and `AbstractFileSetCheck` as they are only a handful.

If we want to create a test around this, we would need some source to verify that fields are indeed properties (like metadata of checks). Though metadata does not exist for the modules mentioned above, we have `config.xml` defining properties for the checker.

For `AbstractCheck` and `AbstractFileSetCheck`, we can get fields defined, and scour the metadata of checks extending them, to find if that field is a check property or not.

This approach is time-consuming as we would have to scan all the subclasses because some fields (which are not properties) are not in any metadata file, so every subclass would have to be traversed.

I don't think the hassle mentioned above is worth it, we rarely modify properties in the modules mentioned above.

As far as listeners are concerned, we don't have any documented properties regarding them. So it can't be judged whether their fields should be annotated or not. 

> I am curious about why you choose an annotation for each type of module instead of a singular one like `ModuleProperty`

That change can be done, I did that just to be explicit but am ready to change it if required.

Reply:
Module property.

Reply:
> Module property.

Done.